### PR TITLE
Add MaximumSubarrayAlgorithm

### PR DESCRIPTION
**Pull request by klevinazeraj**
- [x ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ x] This pull request is all my own work -- I have not plagiarized it.
- [x ] All filenames are in PascalCase.
- [x ] All functions and variable names follow Java naming conventions.
- [x ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by siriak** [comment]
This algorithm is implemented here https://github.com/TheAlgorithms/Java/blob/1551b8f50ba485b5a7d34fe9ed1fbf650134fb9f/src/main/java/com/thealgorithms/dynamicprogramming/KadaneAlgorithm.java
Please make desired changes there

**Comment by klevinazeraj** [comment]
> This algorithm is implemented here https://github.com/TheAlgorithms/Java/blob/1551b8f50ba485b5a7d34fe9ed1fbf650134fb9f/src/main/java/com/thealgorithms/dynamicprogramming/KadaneAlgorithm.java
> Please make desired changes there

This algorithm falls under divide and conquer category, are you sure?

**Comment by siriak** [comment]
Why is it divide and conquer? I'd say that it's dynamic programming because it just goes over elements 1 by 1 and doesn't explicitly divide input in parts.

**Comment by klevinazeraj** [comment]
It cuts down the array in half's the same way merge sort does and then finds the maximum subarray for each of those subarrays and it returns the greatest one. Here is a great video to visualize the process. https://youtu.be/3GD-3UZGsVI

**Comment by siriak** [comment]
I see, it differs from what've linked to

File: src/main/java/com/thealgorithms/divideandconquer/MaximumSubarrayAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Finds the maximum subarray within a given range of indices.
+     *
+     * @param A the array to search for the maximum subarray
+     * @param low the lower index of the subarray to search
+     * @param high the upper index of the subarray to search
+     * @return an array containing the indices and sum of the maximum subarray within the given range
+     */
+    public static int[] findMaximumSubarray(int[] A, int low, int high) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/divideandconquer/MaximumSubarrayAlgorithm.java:59]: Please add a public function `int[] findMaximumSubarray(int[] array)` that only accepts array and returns only starting and ending indexes of the maximum subarray. And make these functions private, as those are implementation and not contract. They can be tested through the public function.

**Review by siriak — CHANGES_REQUESTED** [review]

**Comment by siddhant2002** [comment]
Can I work on this issue?

**Comment by siddhant2002** [comment]
ok

**Pull request closed without merging**